### DISCUSSION TITLE
RuntimeError: not a dict (null) is raised when use  Document.insert_pdf

### DISCUSSION THREAD

**Discussion by hgdhot** [pymupdf/PyMuPDF#2485] [Q&A]
_**RuntimeError: not a dict (null) is raised when insert_pdf**_

## Describe the bug (mandatory)
When use Document.insert_pdf to insert some pages from another docuemnt, the RuntimeError is raised

## To Reproduce (mandatory)
```
import fitz
src_doc = fitz.open('./non-dict.pdf')
doc = fitz.open()
doc.insert_pdf(src_doc, from_page=0, to_page=9)
doc.save('./non-dict-part.pdf')
```
[non-dict.pdf](https://github.com/pymupdf/PyMuPDF/files/11797395/non-dict.pdf)

## Your configuration (mandatory)
3.8.13 (default, Mar 28 2022, 11:38:47)
[GCC 7.5.0]
 linux

PyMuPDF 1.22.3: Python bindings for the MuPDF 1.22.0 library.
Version date: 2023-05-10 00:00:01.
Built for Python 3.8 on linux (64-bit).

**Comment by JorjMcKie** [comment]
This is not a bug, but data errors in your input PDF `non-dict.pdf`. The list of annotations of page 2 contains non-existing xrefs.
If you skip annotations when copying like this `doc.insert_pdf(src_doc, from_page=0, to_page=9, annots=False)` everything works.

**Reply by JorjMcKie** [reply]
In the next version, we will however improve fault tolerance and check for / skip invalid items in the annotation arrays of source pages.
